File: python/chatbot.py
import google.generativeai as genai
from dotenv import load_dotenv
import os

# Load API key from .env
load_dotenv()
genai.configure(api_key=os.getenv("GEMINI_API_KEY"))

# Create the model
model = genai.GenerativeModel("gemini-1.5-flash")

def chat():
    print("Gemini Chatbot (Python). Type 'exit' to quit.\n")

    history = []  # store conversation history

    while True:
        user_input = input("You: ")

        if user_input.lower().strip() == "exit":
            break

        # Add to history
        history.append({"role": "user", "parts": user_input})

        response = model.generate_content(
            history  # send full conversation context
        )

        bot_reply = response.text
        print("Bot:", bot_reply)

        # Add bot output to history
        history.append({"role": "model", "parts": bot_reply})

if __name__ == "__main__":
    chat()

File: javascript/chatbot.js
import 'dotenv/config';
import readline from 'readline';
import { GoogleGenerativeAI } from "@google/generative-ai";

const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY);
const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let history = [];

function askUser() {
  rl.question("You: ", async (userInput) => {
    if (userInput.toLowerCase().trim() === "exit") {
      rl.close();
      return;
    }

    history.push({ role: "user", parts: userInput });

    const result = await model.generateContent(history);
    const reply = result.response.text();

    console.log("Bot:", reply);

    history.push({ role: "model", parts: reply });

    askUser();
  });
}

console.log("Gemini Chatbot (JavaScript). Type 'exit' to quit.\n");
askUser();
